src/client/sandbox/storages/wrapper.js

```diff
@@ -0,0 +1,209 @@
+        this.lastState = this._getCurrentState();
+
+        window.setInterval(() => this._checkStorageChanged(), 10);
+    }
+
+    _getInitialProperties () {
+        var properties = ['length'];
+
+        for (var property in this) {
+            if (this.hasOwnProperty(property))
+                properties.push(property);
+        }
+
+        return properties;
+    }
+
+    _checkStorageChanged () {
+        var currentState = this._getCurrentState();
+
+        for (var i = 0; i < this.lastState[KEY].length; i++) {
+            var lastStateKey   = this.lastState[KEY][i];
+            var lastStateValue = this.lastState[VALUE][i];
+
+            var keyIndex = currentState[KEY].indexOf(lastStateKey);
+
+            if (keyIndex !== -1) {
+                if (currentState[VALUE][keyIndex] !== lastStateValue)
+                    this._raiseStorageChanged(currentState[KEY][keyIndex], lastStateValue, currentState[VALUE][keyIndex]);
+
+                currentState[KEY].splice(keyIndex, 1);
+                currentState[VALUE].splice(keyIndex, 1);
+            }
+            else
+                this._raiseStorageChanged(lastStateKey, lastStateValue, null);
+        }
+
+        for (var j = 0; j < currentState[KEY].length; j++)
+            this._raiseStorageChanged(currentState[KEY][j], this.EMPTY_OLD_VALUE_ARG, currentState[VALUE][j]);
+
+        this.lastState = this._getCurrentState();
+    }
+
+    _getCurrentState () {
+        var addedProperties = this._getAddedProperties();
+        var state           = [[], []];
+
+        for (var i = 0; i < addedProperties.length; i++) {
+            state[KEY].push(addedProperties[i]);
+            state[VALUE].push(this[addedProperties[i]]);
+        }
+
+        return state;
+    }
+
+    _getAddedProperties () {
+        var properties = [];
+
+        for (var property in this) {
+            if (this.hasOwnProperty(property) && this.initialProperties.indexOf(property) === -1)
+                properties.push(property);
+        }
+
+        return properties;
+    }
+
+    _castToString (value) {
+        // NOTE: The browser automatically translates the key and the value to a string. To repeat this behavior,
+        // we use native storage:
+        // localStorage.setItem(null, null) equivalently to localStorage.setItem('null', 'null')
+        this.nativeStorage[STORAGES_SANDBOX_TEMP] = value;
+
+        return this.nativeStorage[STORAGES_SANDBOX_TEMP];
+    }
+
+    _raiseStorageChanged (key, oldValue, newValue) {
+        var url = destLocation.get();
+
+        this.emit(this.STORAGE_CHANGED_EVENT, { key, oldValue, newValue, url });
+    }
+
+    _loadFromNativeStorage () {
+        var storage = this.nativeStorage[this.nativeStorageKey];
+
+        storage = JSON.parse(storage || '[[],[]]');
+
+        for (var i = 0; i < storage[KEY].length; i++)
+            this[storage[KEY][i]] = storage[VALUE][i];
+    }
+
+    saveToNativeStorage () {
```

Comment:
is this a public method? If so maybe we should it to the `// API` section below?